pkg/prometheus/operator.go

```diff
@@ -2157,3 +2163,11 @@ func validateRelabelConfig(rc monitoringv1.RelabelConfig) error {
 	}
 	return nil
 }
+
+func validateProberURL(url string) bool {
+	if strings.Contains(url, ":") {
+		hostPort := strings.Split(url, ":")
+		return govalidator.IsHost(hostPort[0]) && govalidator.IsPort(hostPort[1])
+	}
+	return govalidator.IsHost(url)
```

Comment:
Taking into account #4481, it would be more convenient if the function returns an error (only one place where the error message is constructed).
Changing the function to return an error instead of a bool could look like:

```suggestion
        hostPort := strings.Split(url, ":")
	if !govalidator.IsHost(hostPort[0]) {
		return errors.Errorf("invalid host: %q", hostPort[0])
	}
	
	if len(hostPort[1]) > 1 && govalidator.IsPort(hostPort[1]) {
		return errors.Errorf("invalid port: %q", hostPort[1])
	}
	
	return nil
```

Reply:
Thank you. I added a case for handling url scheme specified as well. Else error may be reported as invalid port for a case like

``` 
      proberSpec: monitoringv1.ProberSpec{
        URL: "https://blackbox-exporter.example.com",
      }
```

Reply:
discussed offline and decided we don't need to handle each case separately so removing handling the url scheme case separately